src/foam/util/AddressUtil.java

```diff
@@ -112,18 +112,28 @@ public static String normalizeCountry(X x, String countryCode) {
       return countryCode;
     }
     
-    String normalizedCountry = countryCode;
+    String[] normalizedCountry = { countryCode };
 
     DAO countryDAO = (DAO) x.get("countryDAO");
-    Country country = (Country) countryDAO.find(OR(
-      EQ(Country.ISO31661CODE, countryCode),
-      EQ(Country.CODE, countryCode)
-    ));
-
-    if ( country != null ) {
-      normalizedCountry = country.getCode();
-    }
+    countryDAO
+      .where(OR(
+        EQ(Country.ISO31661CODE, countryCode),
+        EQ(Country.CODE, countryCode),
+        STARTS_WITH_IC(Country.NAME, countryCode)
+      ))
+      .select(new AbstractSink() {
+        public void put(Object o, Detachable d) {
+          Country country = (Country) o;
+          if (
+            country.getIso31661Code().equals(countryCode) ||
+            country.getCode().equals(countryCode) ||
+            country.getName().equalsIgnoreCase(countryCode) ) {
```

Comment:
AlternativeNames should be checked. 
I believe StringArray can be indexed, and/or IN works can work against the alternativeNames array. 
IN(countryCode, Country.ALTERNATIVE_NAMES)

Reply:
This won't match 'CA' for Canada .. is there a plan to allow for that?